### ISSUE TITLE
HPO with custom Pendulum Environment

### ISSUE THREAD

**Issue by meric-sakarya** [hill-a/stable-baselines#983]
I am working with a customized Pendulum-v0 environment which has a maximum timestep attribute to allow early resets and therefore reward tracking and also uses rgb-array's for observation. I am trying to use the hyperparameter optimization for that environment but could not find any good documentation on how to use Optuna for custom environments this is the code for the custom environment. I already checked [araffin/rl-baselines-zoo#29](https://github.com/araffin/rl-baselines-zoo/issues/29) but I could not understand how to apply it to my case. I am also adding the file configuration on my pc.
 
The code:  

```# modified from https://github.com/openai/gym/blob/master/gym/envs/classic_control/pendulum.py

import gym
from gym import spaces
from gym.utils import seeding
import numpy as np
from os import path

from skimage.color import rgb2gray


class PendulumEnv(gym.Env):
    metadata = {
        'render.modes': ['human', 'rgb_array'],
        'video.frames_per_second': 30
    }

    def __init__(self, g=10.0, max_ts=200):
        self.max_speed = 8
        self.max_torque = 2.
        self.use_images = True  # Default state or Image
        self.dt = .05
        self.g = g
        self.m = 1.
        self.l = 1.
        self.viewer = None
        self.max_ts = max_ts
        self.ts = 0

        high = np.array([1., 1., self.max_speed], dtype=np.float32)
        self.action_space = spaces.Box(
            low=-self.max_torque,
            high=self.max_torque, shape=(1,),
            dtype=np.float32
        )
        if not self.use_images:
            self.observation_space = spaces.Box(
                low=-high,
                high=high,
                dtype=np.float32
            )
        else:
            self.observation_space = spaces.Box(
                low=0, high=255,
                shape=(50, 50, 1),
                dtype=np.uint8)

        self.seed()

    def seed(self, seed=None):
        self.np_random, seed = seeding.np_random(seed)
        return [seed]

    def step(self, u):
        th, thdot = self.state  # th := theta

        g = self.g
        m = self.m
        l = self.l
        dt = self.dt

        u = np.clip(u, -self.max_torque, self.max_torque)[0]
        self.last_u = u  # for rendering
        costs = angle_normalize(th) ** 2 + .1 * thdot ** 2 + .001 * (u ** 2)

        newthdot = thdot + (-3 * g / (2 * l) * np.sin(th + np.pi) + 3. / (m * l ** 2) * u) * dt
        newth = th + newthdot * dt
        newthdot = np.clip(newthdot, -self.max_speed, self.max_speed)

        self.ts += 1

        terminal = False
        if self.ts > self.max_ts:
            terminal = True

        self.state = np.array([newth, newthdot])
        return self._get_obs(), -costs, terminal, {}

    def reset(self):
        self.ts = 0
        high = np.array([np.pi / 2., 1])  # SIMPLIFIED
        self.state = self.np_random.uniform(low=-high, high=high)
        self.last_u = None
        return self._get_obs()

    def _get_obs(self):
        if self.use_images:
            img = self.render('rgb_array')
            img = img[0:50, 0:50, 0]
            # gray_img = gray_img < 0.45
            # gray_img = gray_img > 0.4
            return img[:, :, np.newaxis]
        else:
            theta, thetadot = self.state
            return np.array([np.cos(theta), np.sin(theta), thetadot])

    def render(self, mode='human'):
        if self.viewer is None:
            from gym.envs.classic_control import rendering
            self.viewer = rendering.Viewer(50, 50)
            self.viewer.set_bounds(-1.2, 1.2, -1.2, 1.2)
            rod = rendering.make_capsule(1, .2)
            rod.set_color(.0, .0, .0)  # CHANGED
            self.pole_transform = rendering.Transform()
            rod.add_attr(self.pole_transform)
            self.viewer.add_geom(rod)
            axle = rendering.make_circle(.05)
            axle.set_color(0, 0, 0)  # CHANGED
            self.viewer.add_geom(axle)
#            fname = path.join(path.dirname(__file__), "assets/clockwise.png")
#            self.img = rendering.Image(fname, 1., 1.)
            self.imgtrans = rendering.Transform()
#            self.img.add_attr(self.imgtrans)

#        self.viewer.add_onetime(self.img)
        self.pole_transform.set_rotation(self.state[0] + np.pi / 2)
        if self.last_u:
            self.imgtrans.scale = (-self.last_u / 2, np.abs(self.last_u) / 2)

        return self.viewer.render(return_rgb_array=mode == 'rgb_array')

    def close(self):
        if self.viewer:
            self.viewer.close()
            self.viewer = None


def angle_normalize(x):
    return ((x + np.pi) % (2 * np.pi)) - np.pi


```

File configuartion:

![2020-08-22 (3)](https://user-images.githubusercontent.com/38434300/90955472-f757e900-e47d-11ea-985a-82adaab54b93.png)

Zoo is in the folder named after itself and Code for the pendulum environment is in a subfolder called Environments inside PyCharm.

**System Info**
Describe the characteristic of your environment:
 * Describe how stable baselines was installed (pip, docker, source, ...): pip install stable-baselines[mpi]
 * GPU models and configuration: NVIDIA GTX 1050
 * Python version: 3.7
 * Tensorflow version: 1.15
